mautic/mautic
# Cannot merge company with itself

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [x]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #12599  



#### Description:
This PR blocks the possibility of merging a company with itself by not showing the company in the merge list.
![image](https://github.com/mautic/mautic/assets/97207366/cbf9fc49-91ec-4d10-9568-92764ac345fd)




#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Go to `Companies` and click on any company.
3. Click on `Merge` (available in the header options).
4. In the `Search options`, verify whether the currently viewed company is present or not.
